botocore/client.py

```diff
@@ -174,8 +171,46 @@ def waiter_names(self):
     def _load_service_model(self, service_name):
         json_model = self._loader.load_service_model('aws/%s' % service_name)
         service_model = ServiceModel(json_model, service_name=service_name)
+        self._register_retries(service_model)
         return service_model
 
+    def _register_retries(self, service_model):
+        endpoint_prefix = service_model.endpoint_prefix
+
+        # First, we load the entire retry config for all services,
+        # then pull out just the information we need.
+        original_config = self._loader.load_data('aws/_retry')
+        if not original_config:
+            return
+
+        retry_config = translate.build_retry_config(
+            endpoint_prefix, original_config.get('retry', {}),
+            original_config.get('definitions', {}))
+
+        logger.debug("Registering retry handlers for service: %s",
+                     service_model.service_name)
+        handler = retryhandler.create_retry_handler(
+            retry_config, endpoint_prefix)
+        unique_id = 'retry-config-%s' % endpoint_prefix
+        self._event_emitter.register('needs-retry.%s' % endpoint_prefix,
+                                     handler, unique_id=unique_id)
+
+        # There's certainly a tradeoff for registering the retry config
+        # for the operations when the service is created.  In practice,
+        # there aren't a whole lot of per operation retry configs so
+        # this is ok for now.
+        for key in retry_config:
```

Comment:
On second thought, do we even have any per operation configs?  I've vote for just removing this functionality altogether if we're not using it.